src/manifest.rs

```diff
@@ -289,14 +301,76 @@ impl Manifest {
     }
 
     /// Loads the manifest from the current directory
-    pub async fn read() -> miette::Result<Self> {
-        Self::try_read_from(MANIFEST_FILE)
+    pub async fn read(config: &config::Config) -> miette::Result<Self> {
+        Self::try_read_from(MANIFEST_FILE, Some(config))
             .await?
             .ok_or(miette!("`{MANIFEST_FILE}` does not exist"))
     }
 
+    /// Parses the manifest from the given string
+    ///
+    /// # Arguments
+    /// * `contents` - The contents of the manifest file
+    /// * `config` - The configuration to use for resolving registry aliases
+    pub fn try_parse(contents: &str, config: Option<&config::Config>) -> miette::Result<Self> {
+        let raw: RawManifest = toml::from_str(contents)
+            .into_diagnostic()
+            .wrap_err(DeserializationError(ManagedFile::Manifest))?;
+
+        let dependencies = raw
+            .dependencies()
+            .iter()
+            .map(|(package, manifest)| {
+                let package = package.clone();
+                let manifest = match manifest {
+                    DependencyManifest::Remote(remote_manifest) => {
+                        // For remote manifest dependencies, resolve the registry alias
+                        DependencyManifest::Remote(RemoteDependencyManifest {
+                            version: remote_manifest.version.clone(),
+                            repository: remote_manifest.repository.clone(),
+                            registry: remote_manifest.registry.with_alias_resolved(config)?,
+                        })
+                    }
+                    DependencyManifest::Local(local_manifest) => {
+                        // For local dependencies, check if a remote manifest is present
+                        // and resolve its registry alias
+                        if let Some(ref remote_manifest) = local_manifest.publish {
+                            DependencyManifest::Local(LocalDependencyManifest {
+                                path: local_manifest.path.clone(),
+                                publish: Some(RemoteDependencyManifest {
```

Comment:
Ie. all of this can be either done in the resolve call or ideally during dependency resolution 

So you would only need to read the config in that part of buffrs instead of everywhere